rfd/0078-role-variables.md

```diff
@@ -0,0 +1,176 @@
+---
+authors: Nic Klaassen (nic@goteleport.com)
+state: draft
+---
+
+# RFD 78 - Role Variables
+
+## What
+
+Teleport admins should be able to define variables within roles which can extend
+claims coming from internal and external user traits.
+
+Requirements:
+
+1. Variable definitions should support some form of pattern matching so that
+   traits can be easily and powerfully extended.
+2. A variable defined in one role should be usable in many other roles.
+
+Non-requirements:
+
+1. Variables can depend on other variables.
+
+## Why
+
+Variables will be useful to specify additional traits that are not available in
+OIDC/SAML claims.
+
+Teleport admins don't always have control over their org's OIDC/SAML provider.
+
+Role variables will simplify role evaluation, keep logic in one place, and
+minimize repetition.
```

Comment:
Can we add a sample use case that illustrates something that is not possible without role variables?

Reply:
A lot is possible with enough roles and creative/repetitive use of `regex.replace`, variables will be great for minimizing repetition though. I will add some more examples.

Reply:
Added a good example in [d41536b](https://github.com/gravitational/teleport/pull/14977/commits/d41536b4df8e83de90de64ae75aebb5659655a14)